spec/requests/topics_controller_spec.rb

```diff
@@ -4061,6 +4061,84 @@ def extract_post_stream
         end
       end
     end
+
+    describe "new and unread" do
+      fab!(:group) { Fabricate(:group) }
+      fab!(:new_topic) { Fabricate(:topic) }
+      fab!(:unread_topic) { Fabricate(:topic, highest_post_number: 3) }
+      fab!(:topic_user) do
+        Fabricate(
+          :topic_user,
+          topic: unread_topic,
+          user: user,
+          notification_level: NotificationLevels.topic_levels[:tracking],
+          last_read_post_number: 1,
+        )
+      end
+
+      before do
+        create_post(topic: unread_topic)
+        create_post(topic: unread_topic)
+        user.groups << group
+        SiteSetting.experimental_new_new_view_groups = group.id
+        sign_in(user)
+      end
+
+      it "dismisses new topics" do
+        put "/topics/reset-new.json"
```

Comment:
Can we assert that the response code is 200 and the response body contains the dismissed topic IDs that we expect?

(I'm leaving just one comment about this, but we should do these assertions for all the other requests like this)